src/blop/utils/sets.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+
+
+def validate_set(s, type="continuous") -> tuple:
+    """
+    Check
+    """
+    if type == "continuous":
+        if len(s) == 2:
+            try:
+                x1, x2 = float(s[0]), float(s[1])
+                if x1 <= x2:
+                    return (x1, x2)
+            except Exception:
+                pass
+        raise ValueError(
+            f"Invalid continuous set {s}; valid continuous sets it must be a tuple of two numbers x1, x2 such that x2 >= x1"
+        )
```

Comment:
Can you add the type casting here again? Also, we probably should raise an exception is the comparison doesn't work.
```suggestion
        if len(s) == 2:
            domain = cast(tuple[float, float], s)
            x1, x2 = domain
            if x1 <= x2:
                return (x1, x2)
        raise ValueError(
            f"Invalid continuous set {s}; valid continuous sets it must be a tuple of two numbers x1, x2 such that x2 >= x1"
        )
```